src/main/java/control/ControllerManager.java

```diff
@@ -146,7 +148,7 @@ private void handleCancel(MouseEvent event) {
      *
      * @param block ShotBlock to set as active
      */
-    public void setActiveShotBlock(ShotBlock block) {
+    public void setActiveShotBlock(Optional<ShotBlock> block) {
```

Comment:
I agree with martijn's point, the setter should be a normal ShotBlock and you should do the null check in the setter